src/ImageRunModal.jsx

```diff
@@ -58,6 +157,8 @@ export class ImageRunModal extends React.Component {
             props.resources.memory = this.state.memory * (1024 ** units[this.state.memoryUnit].base1024Exponent);
         if (this.state.hasTTY)
             props.tty = true;
+        if (this.state.publish.length > 0)
+            props.publish = this.state.publish;
```

Comment:
A React component's `props` should never be modified from inside the class. These are read-only, and meant to only be changed from outside (in which case the component re-renders). Also, why duplicate it if you already track it in state?

Reply:
this is rather confusing but props here is a local variable passed to the container create.

Reply:
Should I also avoid aliasing here? So that `getCreateConfig()` actually returns a snapshot?